src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -248,11 +247,22 @@ ucc_status_t ucc_tl_ucp_reduce_scatter_knomial_init_r(
     ucc_assert(coll_args->args.src.info.mem_type ==
                coll_args->args.dst.info.mem_type);
     ucc_knomial_pattern_init(size, rank, radix, &task->reduce_scatter_kn.p);
+    task->reduce_scatter_kn.scratch_mc_header = NULL;
 
+    if (KN_NODE_EXTRA == task->reduce_scatter_kn.p.node_type) {
+        max_recv_size = data_size;
+    } else {
+        step_radix =
+            ucc_sra_kn_compute_step_radix(rank, size,
+                                          &task->reduce_scatter_kn.p);
+        max_recv_size = ucc_sra_kn_compute_seg_size(count, step_radix, 0) *
+            step_radix * dt_size;
+    }
     if (UCC_IS_INPLACE(coll_args->args) ||
-        (KN_NODE_PROXY == task->reduce_scatter_kn.p.node_type)) {
+        (KN_NODE_PROXY == task->reduce_scatter_kn.p.node_type) ||
```

Comment:
just two questions
1. Why condition is (UCC_IS_INPLACE or node_type == PROXY or max_recv_size > data_size), should it be 
((UCC_IS_INPLACE and node_type == PROXY) or max_recv_size > data_size)
2. EXTRA rank doesn't need any additional space right? Maybe add it to condition

Reply:
Regarding (2) - argee. Extra never needs scratch. Will fix.
Regarding (1) - no. PROXY always needs scratch. If UCC_IN_PLACE both PROXY and BASE need scratch. If rbuf is too small (max_recv_size > data_size) also always need bigger scratch. In other words: we can avoid scratch only if: (NON_INPLACE && max_recv_size <= data_isze && NOT_PROXY) || EXTRA.